astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,399 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import re
+import warnings
+from collections import defaultdict, OrderedDict
+
+import numpy as np
+
+from . import Header, Card
+
+from ...coordinates import EarthLocation
+from ...table import Column
+from ...time import Time
+from ...time.core import BARYCENTRIC_SCALES
+from ...time.formats import FITS_DEPRECATED_SCALES
+from ...utils.exceptions import AstropyUserWarning
+
+# The following is based on the FITS WCS Paper IV, "Representations of time
+# coordinates in FITS".
+# http://adsabs.harvard.edu/abs/2015A%26A...574A..36R
+
+
+# FITS WCS standard specified "4-3" form for non-linear coordinate types
+TCTYP_RE_TYPE = re.compile(r'(?P<type>[A-Z]+)[-]+')
+TCTYP_RE_ALGO = re.compile(r'(?P<algo>[A-Z]+)\s*')
+
+
+# FITS Time standard specified time units
+FITS_TIME_UNIT = ['s', 'd', 'a', 'cy', 'min', 'h', 'yr', 'ta', 'Ba']
+
+
+# Global time reference coordinate keywords
+TIME_KEYWORDS = ('TIMESYS', 'MJDREF', 'JDREF', 'DATEREF',
+                 'TREFPOS', 'TREFDIR', 'TIMEUNIT', 'TIMEOFFS',
+                 'OBSGEO-X', 'OBSGEO-Y', 'OBSGEO-Z', 'DATE',
+                 'DATE-OBS', 'DATE-AVG', 'DATE-BEG', 'DATE-END',
+                 'MJD-OBS', 'MJD-AVG', 'MJD-BEG', 'MJD-END')
+
+
+# Column-specific time override keywords
+COLUMN_TIME_KEYWORDS = ('TCTYP', 'TCUNI', 'TRPOS')
+
+
+# Column-specific keywords regex
+COLUMN_TIME_KEYWORD_REGEXP = '({0})[0-9]+'.format(
+    '|'.join(COLUMN_TIME_KEYWORDS))
+
+
+def is_time_column_keyword(keyword):
+    """
+    Check if the FITS header keyword is a time column-specific keyword.
+
+    Parameters
+    ----------
+    keyword : str
+        FITS keyword.
+    """
+    return re.match(COLUMN_TIME_KEYWORD_REGEXP, keyword) is not None
+
+
+# Set astropy time global information
+GLOBAL_TIME_INFO = {'TIMESYS': ('UTC', 'Default time scale'),
+                    'JDREF': (0.0, 'Time columns are jd = jd1 + jd2'),
+                    'TREFPOS': ('TOPOCENTER', 'Time reference position')}
+
+
+def _verify_global_info(global_info):
+    """
+    Given the global time reference frame information, verify that
+    each global time coordinate attribute will be given a valid value.
+
+    Parameters
+    ----------
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Translate FITS deprecated scale into astropy scale, or else just convert
+    # to lower case for further checks.
+    global_info['TIMESYS'] = FITS_DEPRECATED_SCALES.get(global_info['TIMESYS'],
```

Comment:
This is somewhat counter to my earlier suggestion, but I think that here I would set a new `scale` entry in the dictionary, i.e.,
```
global_info['scale'] = ...
```
and then below use `format` - I think this more clearly separates what comes from the header and what is inferred. Obviously, elsewhere you then have to use `scale` and `format`.